test/cli/cli_tests.ml

```diff
@@ -0,0 +1,601 @@
+(** Unit tests to ensure CLI parameter format does not change.
+*)
+
+type setting_value = [
+  `B of string * bool ref * bool
+  | `I of string * int ref * int
+  | `F of string * float ref * float
+  | `S of string * string ref * string
+  | `SO of string * string option ref * string option
+  | `SL of string * string list ref * string list
+  | `ML of string * Tlapm_lib__Method.t list ref * Tlapm_lib__Method.t list
+]
+
+let capture_output (f : Format.formatter -> unit) : string =
+  let out = Buffer.create 0 in
+  let out_format = Format.formatter_of_buffer out in
+  f out_format;
+  Format.pp_print_flush out_format ();
+  Buffer.contents out
+  
+
+let rec string_contains (needle : string) (haystack : string) : bool =
+  match haystack with
+  | "" -> false
+  | _ -> String.starts_with ~prefix:needle haystack ||
+    String.sub haystack 1 ((String.length haystack) - 1) |>
+    string_contains needle
```

Comment:
This allocates a lot of temporary strings, since String.sub returns a copy of the substring.
Could you add a comment to warn against using this function on large haystacks ?